### DISCUSSION TITLE
Paste Clipboard Images

### DISCUSSION THREAD

**Discussion by px6** [lllyasviel/stable-diffusion-webui-forge#1781] [Ideas]
It would significantly enhance the workflow if we could paste clipboard images directly into any image-relevant window on the page. This feature was available in A1111 and was one of its underrated features.

I discovered that something similar can be done in Forge by using the enlarge icon and then pasting, but from a UX standpoint, that process takes longer, also, pressing the escape key doesn’t close the enlarged image preview. It would be much faster to allow pasting directly into the prompt input fields tied to the image being pasted.

You could easily copy any image generated from txt2img or grab one quickly from the web and paste it wherever needed within the Forge web interface... whether for upscaling, img2video, or any img2img tab (inpaint, sketch, etc.).

![image](https://github.com/user-attachments/assets/5370bd3a-9af8-444f-95e9-6c4566f7d5d0)

Thank you for contributing to the development of this Web UI!!!

best regards.
Pax

**Comment by slipperyjankinz** [comment]
I came here to suggest the same thing.  It'll be great from a workflow standpoint.

Fooocus was similar and [some guy made a quick script to allow copy/paste](https://github.com/lllyasviel/stable-diffusion-webui-forge/discussions/1796). It worked great.  I hope image copy/paste can be added to Forge.  I use images from various sources and having to navigate through save/load folder is quite the slowdown for iterative work.

**Comment by DenOfEquity** [comment]
This already works for me. You need to mouseover the target, then Ctrl+V will paste into it. Drag-drop works too.

**Reply by slipperyjankinz** [reply]
Hey, what browser are you using?  I'm using Brave, but will to switch to one that works.

Drag and drop from Windows Explorer or another web page works, but pasting images from anywhere is the real time saver.

**Reply by DenOfEquity** [reply]
Edge, also tested Firefox successfully.

**Reply by Vagrin** [reply]
Works fine for me in Chrome. And yes... very handy.
